h2/src/main/org/h2/command/CommandRemote.java

```diff
@@ -212,28 +213,28 @@ public ResultWithGeneratedKeys executeUpdate(Object generatedKeysRequest) {
                     transfer.writeInt(SessionRemote.COMMAND_EXECUTE_UPDATE).writeInt(id);
                     sendParameters(transfer);
                     if (supportsGeneratedKeys) {
-                        if (Boolean.FALSE.equals(generatedKeysRequest)) {
-                            transfer.writeInt(0);
+                        int mode = GeneratedKeysMode.valueOf(generatedKeysRequest);
+                        transfer.writeInt(mode);
+                        switch (mode) {
+                        case GeneratedKeysMode.NONE:
```

Comment:
Isn't this case a NOOP, like the AUTO case, because readGeneratedKeys is already false?

Reply:
Yes, but it may be used as a guard against illegal values of `generatedKeysRequest`. However, this parameter is always generated internally, so this case is not very useful.

However, I add code that throws exception to `GeneratedKeysMode.valueOf()` so illegal values of this parameter (if someone changes the sources in the wrong way) can be detected.